### ISSUE TITLE
[USER ISSUE] Disappearing dependency files in CodaLab runs

### ISSUE THREAD

**Issue by nelson-liu** [codalab/codalab-worksheets#1417]
**Describe the bug**

I'm having a strange issue where CodaLab will fail to find files that live in dependencies of the run bundle. Running `tree` on the filesystem shows that the file exists, but when I try to use it later in the run, bash throws a `No such file or directory`. The error seems non-deterministic, and simply rerunning the bundle solves the issue in most cases (but this is annoying, since the bundle involves training a model). Maybe there's something about the fact that I'm running several of these bundles that rely on the same dependency at once, and there's some race condition with mounting?

**Codalab Information**
- *Your username*: nfliu
- *UUID of worksheet where problem occurred*: 0xa49368ebd6e64154b683ed8812634ae9 (private worksheet, let me know who I should give access to)
- *UUID of problem bundle (please don't delete)*: 0xf41d176a08124c609a7cf5efe7d50ae8
- *Error message received from Codalab (if applicable)*: `bash: ./code/teardown_cl_runtime.sh: No such file or directory`
 - *OS (if on CLI)*: Mac OS Mojave
 - *Codalab CLI Version (if on CLI)*: 0.4.0 
 - *Browser (if on Web)*; Chrome

**To Reproduce**
Steps to reproduce the behavior :
1. I'm not really sure how to reliably reproduce this problem on a particular bundle, but if many experiments, usually at least one fails.

**Expected behavior**

The file should be found. Note that earlier in the run, I call the `tree` command to print out the file system hierarchy, and the file is indeed there. The `allennlp train ...` command should not remove anything, so I still expect it to be there.

**Additional context**

The run that fails with `bash: ./code/teardown_cl_runtime.sh: No such file or directory` is exactly the same as the run that succeeds here ( worksheets.codalab.org/bundles/0xc68db226d5fa48228d0bb60f318c47e3 ), save for the fact that the names are different. Otherwise, things are the same, and I'd expect them to either both fail, or both succeed.

Also, not sure if it matters, but I'm running on the SLURM cluster with `nlp-cl-worker`

Sorry, this is another vague issue, but I'm not quite sure where to begin looking for the culprit...I'll rerun the failing bundle above and show that it succeeds under the exact same conditions.

**Comment by nelson-liu** [comment]
The issue went away, it might be an odd interaction with disk quota? Feel free to close.

**Comment by nelson-liu** [comment]
Nevermind, the problem occurs even when my disk quota is more than large enough to store the output of all ongoing runs. See https://worksheets.codalab.org/bundles/0xf41d176a08124c609a7cf5efe7d50ae8

**Comment by nelson-liu** [comment]
Ok, here's a pair of bundles that demonstrates the issue. The only thing that differs between the bundles is the name of the run, all dependencies are otherwise the same, and the run command is the same as well.

This run failed: https://worksheets.codalab.org/bundles/0xf41d176a08124c609a7cf5efe7d50ae8

This run succeeded: https://worksheets.codalab.org/bundles/0xc68db226d5fa48228d0bb60f318c47e3

**Comment by percyliang** [comment]
Does this happen consistently?

**Comment by nelson-liu** [comment]
No, I haven't figured out how to reliably reproduce this. I'm running ~10 experiments in parallel, and usually ~3-5 of them run into this.

**Comment by epicfaace** [comment]
I'd imagine this is also a timing issue that causes dependencies to be deleted, as I describe in https://github.com/codalab/codalab-worksheets/pull/3669. I'm not sure if https://github.com/codalab/codalab-worksheets/pull/3669 fixes this, because this issue was reported (in September 2019) before Tony's original PR https://github.com/codalab/codalab-worksheets/pull/2295 from May 2020 -- but maybe let's close this PR once the fix is merged and reopen if we still see this issue happening.

**Comment by epicfaace** [comment]
Going to close this as a duplicate of https://github.com/codalab/codalab-worksheets/issues/3627, then.
